### ISSUE TITLE
Autocomplete color not changing back to default after accepting

### ISSUE THREAD

**Issue by mankinskin** [zsh-users/zsh-autosuggestions#559]
### Describe the bug
When I get a suggestion and accept it via right arrow key, the color remains as the autocomplete style, instead of changing back to default.

### To Reproduce
```
# ~/.zshrc
ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE="fg=#888888, bg=black,bold"
ZSH_AUTOSUGGEST_STRATEGY=(match_prev_cmd completion history)
set ZSH_AUTOSUGGEST_USE_ASYNC
```
type ".zsh" in home directory (or provoke any autosuggestion)
see the autocompletion in gray
hit right arrow key
see the completed string still in gray

### Expected behavior
the completed string should be the default text color

### Screenshots
![autosuggest](https://user-images.githubusercontent.com/20745737/90314547-d854d680-df14-11ea-94e8-b8e8ced38dd9.png)

**Comment by mankinskin** [comment]
A workaround is to also install [zsh-syntax-highlighting](https://github.com/zsh-users/zsh-syntax-highlighting), as this overrides the colors.

**Comment by Leander-van-Boven** [comment]
I have installed zsh-syntax-highlighting, which works, but I still have the same bug.

**Comment by Leander-van-Boven** [comment]
To provide more information:

I encountered the same problem as described in this issue, although I set the following in my .zshrc:
```bash
ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE="fg=015"
```

Then I installed zsh-syntax-highlighting as suggested. This seems to only solve the problem partially. Certain (key)words get replaced by both color and style and thus you won't notice this happening. However other parts of the command keep their ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE foreground color even after accepting the suggestion.

Example below, in the first command I did not set the ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE environment variable, for the second I set it again to the value above. In both cases I accepted the suggestion after typing `so`.

<img width="158" height="103" alt="Image" src="https://github.com/user-attachments/assets/036b703c-8524-4ed9-98cc-5863d515360e" />
